spring-kafka/src/test/java/org/springframework/kafka/support/resolver/ResolvableTypeResolverTests.java

```diff
@@ -0,0 +1,165 @@
+		final String s = new ObjectMapper().writeValueAsString(new Foo().setText(TEXT));
+		final Map<String, Object> contentType = Map.of(MessageHeaders.CONTENT_TYPE, "application/xml");
+		// No converters can process xml media type, so will cause MessageConversionException.
+		assertThatThrownBy(() ->
+				method.invoke(new GenericMessage<>(s, contentType)))
+				.isExactlyInstanceOf(MessageConversionException.class);
+		// Ignore the header, and force using own converter.
+		InvocableHandlerMethod forceMethod = getForceJsonFooInvocableHandlerMethod();
+		forceMethod.invoke(new GenericMessage<>(s, contentType));
+	}
+
+	public void onForceJsonMessageWithFoo(@ResolvableType(force = true) Foo payload) {
+		assertThat(payload.getText()).isEqualTo(TEXT);
+	}
+
+	@NotNull
+	private InvocableHandlerMethod getFooInvocableHandlerMethod() throws NoSuchMethodException {
+		MessageHandlerMethodFactory factory = getMessageHandlerMethodFactory();
+		return factory.createInvocableHandlerMethod(this, getClass().getDeclaredMethod(
+				"onMessageWithFoo", Foo.class));
+	}
+
+	@NotNull
+	private InvocableHandlerMethod getForceJsonFooInvocableHandlerMethod() throws NoSuchMethodException {
+		MessageHandlerMethodFactory factory = getMessageHandlerMethodFactory();
+		return factory.createInvocableHandlerMethod(this, getClass().getDeclaredMethod(
+				"onForceJsonMessageWithFoo", Foo.class));
+	}
+
+	@SuppressWarnings("rawtypes")
+	private MessageHandlerMethodFactory getMessageHandlerMethodFactory() {
+		KafkaListenerAnnotationBeanPostProcessor bpp = new KafkaListenerAnnotationBeanPostProcessor<>();
+		beanPostProcessorConfig(bpp);
+		return bpp.getMessageHandlerMethodFactory();
+	}
+
+	@SuppressWarnings("rawtypes")
+	private void beanPostProcessorConfig(KafkaListenerAnnotationBeanPostProcessor processor) {
+		processor.setBeanFactory(new KafkaBeanFactory());
+		try {
+			processor.afterSingletonsInstantiated();
+		} catch (NoSuchBeanDefinitionException ignore) {
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	static class KafkaBeanFactory extends DefaultListableBeanFactory {
+		@Override
+		public <T> Map<String, T> getBeansOfType(Class<T> type) throws BeansException {
+			return new HashMap<>() {
+				{
+					if (KafkaListenerConfigurer.class.equals(type)) {
+						put("bean", (T) new MediaMessageConfig());
+					}
+				}
+			};
+		}
+	}
+
+	static class MediaMessageConfig implements KafkaListenerConfigurer {
+		public CompositeMessageConverter createMessageConverter() {
+			List<MessageConverter> converters = new ArrayList<>();
+			converters.add(new StringMessageConverter());
+			converters.add(new ByteArrayMessageConverter());
+			converters.add(createJacksonConverter());
+			return new CompositeMessageConverter(converters);
+		}
+
+		protected MappingJackson2MessageConverter createJacksonConverter() {
+			MappingJackson2MessageConverter converter = new MappingJackson2MessageConverter();
+			converter.setContentTypeResolver(new DefaultContentTypeResolver());
```

Comment:
It is just a custom usage, If needed, it can change.

Reply:
By the way, This code snippet is just a copy from `org.springframework.messaging.simp.config.AbstractMessageBrokerConfiguration#createJacksonConverter`(Line 474)

Reply:
This is test code. I believe the test code should be as simple as  possible.
It must answer questions, but not raise them.
More over your reference is misleading: you don't set a `defaultMimeType` like it is done in that `AbstractMessageBrokerConfiguration`:
```
protected MappingJackson2MessageConverter createJacksonConverter() {
		DefaultContentTypeResolver resolver = new DefaultContentTypeResolver();
		resolver.setDefaultMimeType(MimeTypeUtils.APPLICATION_JSON);
		MappingJackson2MessageConverter converter = new MappingJackson2MessageConverter();
		converter.setContentTypeResolver(resolver);
		return converter;
	}
```

Reply:
I agree that the test code should be as simple as possible, but it seems that this test case really need to verify some media type like `application/json` can be correctly process by the converter which support for converting it.
Set a `defaultMimeType` is not necessary, since the CONTENT_TYPE header will not be null at all.
(annotation type attribute can't be null, because null is not a constant value)
See `org.springframework.messaging.converter.DefaultContentTypeResolver#resolve`(Line 62)
and with the logic:
```
if (header.CONTENT_TYPE == null) {
  header.CONTENT_TYPE = ann.type();
}
```